src/main/java/org/apache/sysds/runtime/util/DataConverter.java

```diff
@@ -307,6 +307,34 @@ public static MatrixBlock readMatrixFromHDFS(ReadProperties prop)
 		return ret;
 	}
 
+	public static List<Integer> convertVectorToIndicesList(MatrixBlock mb)
+	{
+		int rows = mb.getNumRows();
+		int cols = mb.getNumColumns();
+		ArrayList<Integer> indices = new ArrayList<>();
+
+		if( mb.getNonZeros() > 0 )
+		{
+			if( mb.isInSparseFormat() )
+			{
+				Iterator<IJV> iter = mb.getSparseBlockIterator();
+				while( iter.hasNext() ) {
+					IJV cell = iter.next();
+					if( cell.getV() != 0.0)
+						indices.add(cell.getI()*cols+cell.getJ());
+				}
```

Comment:
on a sparse block call:

```java
SparseBlock sb = mb.getSparseBlock()
int[] aix = sb.indexes(0);
return aix;
```

This is assuming that you have only one row of values.